### PR TITLE
added nb

### LINKED ISSUE #15: Text Classification using Transformers
The idea is to make a new beginner-level NLP tutorial combining this [transformers example](https://github.com/pytorch/ignite/tree/master/examples/contrib/transformers) and this [Text CNN notebook](https://github.com/pytorch/ignite/blob/master/examples/notebooks/TextCNN.ipynb).

### PR DESCRIPTION
**Pull request by Priyansi**
Resolves #15

### CODE REVIEW

File: tutorials/transformers-text-classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,751 @@
...
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "enczLgLTznkH"
+      },
+      "source": [
+        "from ignite.utils import manual_seed\n",
+        "\n",
+        "SEED = 42\n",
+        "manual_seed(SEED)"
```
Comment (sdesrozis) [inline tutorials/transformers-text-classification.ipynb:103]: Matter of taste
```suggestion
        "manual_seed(seed=42)"
```
and remove the line above.

File: tutorials/transformers-text-classification.ipynb [hunk 2]
```diff
@@ -0,0 +1,751 @@
...
+        "Before we dive in, we will seed everything using [`manual_seed`](https://pytorch.org/ignite/utils.html#ignite.utils.manual_seed)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "enczLgLTznkH"
+      },
+      "source": [
+        "from ignite.utils import manual_seed\n",
```
Comment (sdesrozis) [inline tutorials/transformers-text-classification.ipynb:100]: I very much appreciated what was done in the CIFAR10 notebook. Why not try to do in the same ? I mean to have an import section gathering all required imports, a bit more comments from native pytorch sections (dataloader, optimizer,  model) and use `idist` when possible (`device`, and launch).
What do you think ?

File: tutorials/transformers-text-classification.ipynb [hunk 3]
```diff
@@ -0,0 +1,751 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0-Yd47-UiLFs"
+      },
+      "source": [
+        "### LR Scheduler\n",
+        "\n",
+        "We will use the built-in Ignite alternative of `linear` scheduler which is [`PiecewiseLinear`](https://pytorch.org/ignite/generated/ignite.handlers.param_scheduler.PiecewiseLinear.html#piecewiselinear). We will also increase the number of epochs."
```
Comment (sdesrozis) [inline tutorials/transformers-text-classification.ipynb:268]: Why not introduce a warmup, it's a cool reason to explain why use this scheduler instead of the PyTorch one.

File: tutorials/transformers-text-classification.ipynb [hunk 4]
```diff
@@ -0,0 +1,751 @@
...
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "A0WXnFD7bg2C"
+      },
+      "source": [
+        "import torch\n",
+        "\n",
+        "device = torch.device(\"cuda\") if torch.cuda.is_available() else torch.device(\"cpu\")\n",
```
Comment (sdesrozis) [inline tutorials/transformers-text-classification.ipynb:310]: Here, use `idist` would be better.

File: tutorials/transformers-text-classification.ipynb [hunk 5]
```diff
@@ -0,0 +1,751 @@
...
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "APNr5lgsygtw"
+      },
+      "source": [
+        "from torch.utils.data import DataLoader\n",
+        "\n",
+        "train_dataloader = DataLoader(small_train_dataset, shuffle=True, batch_size=8)\n",
```
Comment (sdesrozis) [inline tutorials/transformers-text-classification.ipynb:210]: Does `auto_dataloader` work fine here ? Yet, I,didn't try

File: tutorials/transformers-text-classification.ipynb [hunk 6]
```diff
@@ -0,0 +1,751 @@
...
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "XVS15FELytIj"
+      },
+      "source": [
+        "from transformers import AutoModelForSequenceClassification\n",
+        "\n",
+        "model = AutoModelForSequenceClassification.from_pretrained(\"bert-base-cased\", num_labels=2)"
```
Comment (sdesrozis) [inline tutorials/transformers-text-classification.ipynb:233]: Same question concerning `auto_model`.

File: tutorials/transformers-text-classification.ipynb [hunk 7]
```diff
@@ -0,0 +1,751 @@
...
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "RIihFsPryvI_"
+      },
+      "source": [
+        "from transformers import AdamW\n",
+        "\n",
+        "optimizer = AdamW(model.parameters(), lr=5e-5)"
```
Comment (sdesrozis) [inline tutorials/transformers-text-classification.ipynb:255]: And last, `auto_optimizer`.

**Comment by sdesrozis** [comment]
@Priyansi Thank you ! I left a few comments, mainly about `idist` and maybe out of the scope. Anyway, it looks good to me !

File: tutorials/transformers-text-classification.ipynb (hunk 5 above)
Comment (KickItLikeShika) [reply tutorials/transformers-text-classification.ipynb:210]: I don't agree that much to use `auto_*` or any distributed things here, as that's not what we want to show in this tutorial, IMO, we should keep things straightforward as possible in this introductory level tutorial

Comment (sdesrozis) [reply tutorials/transformers-text-classification.ipynb:210]: Transformers are not cheap models, training is time consuming. Although I understand your meaning, auto helpers are very light and could be a valuable improvement. 

Maybe it could be discussed later.

Comment (Priyansi) [reply tutorials/transformers-text-classification.ipynb:210]: Thank you for your review @sdesrozis but I think I agree with @KickItLikeShika . This tutorial is just meant for introducing Ignite concepts for NLP beginners (that's why I used the most basic example by :hugs: ) and `idist` is still an advanced concept that needs to be discussed in detail like in the CIFAR10 notebook (how to use different configs). I can still introduce the `auto_*` methods without going into detail and just explaining how to run this code as a Jupyter notebook or in Colab. But I feel that this will be incomplete. What do you all think?

File: tutorials/transformers-text-classification.ipynb (hunk 3 above)
Comment (Priyansi) [reply tutorials/transformers-text-classification.ipynb:268]: I'm trying to convert the code provided in the reference tutorial:
```f
rom transformers import get_scheduler

num_epochs = 3
num_training_steps = num_epochs * len(train_dataloader)
lr_scheduler = get_scheduler(
    "linear",
    optimizer=optimizer,
    num_warmup_steps=0,
    num_training_steps=num_training_steps
)```
and therefore I didn't introduce a warmup.

File: tutorials/transformers-text-classification.ipynb (hunk 2 above)
Comment (Priyansi) [reply tutorials/transformers-text-classification.ipynb:100]: Since I mentioned at the beginning of the tutorial that we were following :hugs: tutorial for reference, I didn't want to repeat code that is already explained there plus the imports are introduced as they are used.
I can however, not follow that tutorial verbatim and add more explanation so that our tutorial is standalone.

File: tutorials/transformers-text-classification.ipynb (hunk 5 above)
Comment (sdesrozis) [reply tutorials/transformers-text-classification.ipynb:210]: It makes sense ;)

File: tutorials/transformers-text-classification.ipynb (hunk 2 above)
Comment (sdesrozis) [reply tutorials/transformers-text-classification.ipynb:100]: I understand your point. No worries about my comment.

File: tutorials/transformers-text-classification.ipynb (hunk 3 above)
Comment (sdesrozis) [reply tutorials/transformers-text-classification.ipynb:268]: Ok, you mentioned your intention to match to the original code, so my comment doesn’t matter.

**Review by sdesrozis — APPROVED** [review]
@Priyansi Thanks, LGTM !

**Comment by vfdev-5** [comment]
@KickItLikeShika , @Ishan-Kumar2 could you please review and approuve this tutorial if it looks good to you. Thanks
